src/components/ProductDetail/ProductDetail.js

```diff
@@ -0,0 +1,111 @@
+import { useSingleProduct } from "../../utils/hooks/useSingleProduct";
+import { useState } from "react";
+import { useParams } from "react-router-dom";
+
+// Import Swiper React components
+import { Swiper, SwiperSlide } from "swiper/react";
+import SwiperCore, { Navigation, Thumbs } from "swiper";
+import "swiper/swiper-bundle.min.css";
+import "swiper/swiper.min.css";
+// install Swiper modules
+SwiperCore.use([Navigation, Thumbs]);
+
+function ProductDetail() {
+  const { productId } = useParams();
+  const { data: product, isLoading } = useSingleProduct(productId);
+  const [thumbsSwiper, setThumbsSwiper] = useState(null);
+
+  console.log("data", product);
+  console.log("isLoading", isLoading);
+  return (
+    <div className="product-detail">
```

Comment:
Technically there is nothing wrong with this component. But its not that easy to find out what corresponds to each section of the product. 
My suggestion is to use the `ProductDetail` component as a container or layout and then create some other components that really create a detail of a product. 
```suggestion
<div className="product-detail">
      {
      isLoading ? 
      (
         <p>Loading...</p>
      ) 
        : 
      (
       <div className="product-detail-container">
         // inside this component you handle its own styles and everything that corresponds to it
         <ProductImage {...onlyNecessaryProps} /> 
         // then you have your "main" product detail info
         <ProductDetailInfo {...onlyNecessaryProps} />
       <div/>
      )
      }
```
And so on, it is much more easy to understand how everything works like this and easy to maintain, if you need to modify for example only the image you don't touch anything that is not related to that. You can go beyond that and create sub components for the information of the product.

Reply:
Remember in modern React it is more common to have every component with the least amount of logic or information (or what the user views) and divide it into multiple different components that try to accomplish a very simple action or render, better known as `Single Responsibility Principle`. 
Later in the bootcamp there is an 'Atomic Design' lecture, if you start searching about that maybe it is easier for you to understand.

Reply:
I honestly did not thouht about that, I could also have a sub component for the Specs table too! I will try to rework all of it by the next deliverable.